Android/app/src/main/java/com/example/instagramclone/Activities/AddPostActivity.java

```diff
@@ -0,0 +1,149 @@
+package com.example.instagramclone.Activities;
+
+import android.annotation.SuppressLint;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.preference.PreferenceManager;
+import android.util.Log;
+import android.view.View;
+import android.widget.EditText;
+import android.widget.ImageView;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.example.instagramclone.Api.FeedApi;
+import com.example.instagramclone.Common.APIClient;
+import com.example.instagramclone.Common.FileUtil;
+import com.example.instagramclone.Models.FeedItemModel;
+import com.example.instagramclone.Models.LocationModel;
+import com.example.instagramclone.R;
+import com.google.android.libraries.places.api.Places;
+import com.google.android.libraries.places.api.model.Place;
+import com.google.android.libraries.places.widget.Autocomplete;
+import com.google.android.libraries.places.widget.model.AutocompleteActivityMode;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+
+import okhttp3.MediaType;
+import okhttp3.MultipartBody;
+import okhttp3.RequestBody;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class AddPostActivity extends AppCompatActivity {
+    private static final String TAG = "AddPostActivity";
+    private static final int PLACE_AUTOCOMPLETE_REQUEST_CODE = 1;
+
+    private ImageView image;
+    private EditText locationText, captionText;
+    private TextView share;
+    private ProgressBar progressBar;
+
+    private Uri imageUri;
+    private FeedApi feedApi;
+
+    private FeedItemModel postItem;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_add_post);
+        Places.initialize(getApplicationContext(), "AIzaSyCm9w5hmifYrImzBrGzt_4UMcRWFteW_P0");
+        initComponents();
+    }
+
+    private void initComponents() {
```

Comment:
There could be some contract like interface or abstract class with methods `initComponents()` and `initData()`